### PR TITLE
Ignore argument check when expected_arguments empty

### LINKED ISSUE #178: `be_delayed` without expected_arguments never matches
From what I can see, this is due to how the jobs are found:

```ruby
        def find_job(method, arguments, &block)
          job = (::Sidekiq::Extensions::DelayedClass.jobs + ::Sidekiq::Extensions::DelayedModel.jobs + ::Sidekiq::Extensions::DelayedMailer.jobs).find do |job|
            yaml = YAML.load(job['args'].first)
            @expected_method_receiver == yaml[0] && @expected_method.name == yaml[1] && (@expected_arguments <=> yaml[2]) == 0
          end

          yield job if block && job
        end
```

If no expected arguments are given, `(@expected_arguments <=> yaml[2]) == 0` will always be false (for jobs that have been enqueued with arguments).

(Also, I'm not sure why there is a spaceship operator here instead of a simple `==`, and why `method` and `arguments` are passed to `find_job` but never used, but that's another topic).

In short, this comparison should be performed only if expected_arguments are given.

That is, if you want this to be coherent with the rest of rspec. Otherwise, you can dismiss this issue :shrug:

### PR DESCRIPTION
**Pull request by teckwan**
Opened in favor of #180 , to fix #178.

### CODE REVIEW

File: spec/rspec/sidekiq/matchers/be_delayed_spec.rb [hunk 1]
```diff
@@ -120,6 +120,10 @@
             Object.delay.nil?
 
             expect(delay_subject.matches? Object.method :nil?).to be true
+
+            Object.delay.is_a? Object
+
+            expect(delay_subject.matchers? Object.method :is_a?).to be true
```
Comment (wspurgin) [inline spec/rspec/sidekiq/matchers/be_delayed_spec.rb:126]: Small typo

```suggestion
            expect(delay_subject.matches? Object.method :is_a?).to be true
```

**Review by wspurgin — CHANGES_REQUESTED** [review]
Thanks for the PR! I'm glad this is getting fixed 🙌 

PR looks good, but there's one small typo in the spec

File: spec/rspec/sidekiq/matchers/be_delayed_spec.rb (hunk 1 above)
Comment (teckwan) [reply spec/rspec/sidekiq/matchers/be_delayed_spec.rb]: My bad, thanks!

**Review by wspurgin — APPROVED** [review]
🚀 Thanks for the fix!

**Pull request merged**